rs/https_outcalls/consensus/src/pool_manager.rs

```diff
@@ -306,6 +299,10 @@ impl CanisterHttpPoolManagerImpl {
                     ));
                 }
 
+                if !active_callback_ids.contains(&share.content.id) {
+                    return Some(CanisterHttpChangeAction::RemoveUnvalidated(share.clone()));
```

Comment:
I think we might have to do this after deferring the validation of shares from nodes ahead of us ([CON-1227](https://dfinity.atlassian.net/browse/CON-1227)), otherwise we might remove shares that could potentially be valid in the future.

Reply:
Yep, I realized this as well once I started the other ticket. Btw, we're only talking about the order of merging, right?

Reply:
Yes, and the order here in code. I think we can only decide if a callback ID is "active" once we caught up to that ID in state.

The idea would be to skip validation of shares for contexts that are in the "future" and make the priority function only pull artifacts that are at most a certain number of heights in the future, like we already do for consensus/ecdsa.

Reply:
@eichhorl are you ok with me just merging this PR with #752 ?

Reply:
Sure 👍